lib/providers/services/http_service.dart

```diff
@@ -456,7 +456,7 @@ class HttpService {
       Map<String, dynamic> body = res.parseJson();
 
       List<dynamic> activeProposals = body["rows"].where((dynamic item) {
-        return item["stage"] == stage && item["status"] == status;
+        return item["stage"] == stage && item["status"] == status && int.parse(item["staked"].split(".")[0]) > 0 ;
```

Comment:
This has a high risk factor of crashing due to: 
Split() on possible null
Out of bounds,
Trying to parse an Int

Also its hard to understand what the whole thing evaluates to and why are we trying to evaluate it. 

Please consider moving the expression `item["stage"] == stage && item["status"] == status && int.parse(item["staked"].split(".")[0]) > 0 ;` to a helper function. 

There you can give it a meaningful name, leave some comments, and safeguard vs potential crashes. 


Example: 

`  /// Check for active proposal.
  /// The light wallet should only show items that have more than 0 staked
  /// As long as there is 0 staked, nobody can actually vote on it
  bool isItActiveProposal(item, String stage, String status) => item["stage"] == stage && item["status"] == status && int.parse(item["staked"].split(".")[0]) > 0;`

^^ Add as many safety checks as you think are needed.

Reply:
... i cant get that last func to format nicely here....

Reply:
Point taken will make helper function that will be nicer 

By way of explanation though 

the contract ensures that staked is an asset and it is either “0.0000 Seeds” or any other number of seeds; the formatting is fixed it’s always decimal with 4 significant digits 

So there is no way this can be 
- empty 
- not a number
- undefined 

Unless the table structure changes in which case it could be all those things.